packages/adminrouter/extra/src/test-harness/tests/test_common.py

```diff
@@ -0,0 +1,25 @@
+import grp
+
+import psutil
+import pytest
+
+
+# A hack to be able to parameterize a test with multiple fixtures.
+# https://github.com/pytest-dev/pytest/issues/349#issuecomment-189370273
+@pytest.fixture(
+    params=[
+        'agent_ar_process',
+        'master_ar_process'
```

Comment:
The problem here is that both fixtures are test-file scoped :( I get something like this:

```
root       128  0.0  0.0  43232  6484 ?        Ss   18:33   0:00  \_ nginx: master process /usr/local/adminrouter/nginx/sbin/nginx -c /usr/local/adminrouter/nginx/conf/nginx.agent.conf -g daemon off;
nobody     129  0.0  0.0  43236  3364 ?        S    18:33   0:00  |   \_ nginx: worker process
root       139  0.0  0.1 152012  8520 ?        Ss   18:33   0:00  \_ nginx: master process /usr/local/adminrouter/nginx/sbin/nginx -c /usr/local/adminrouter/nginx/conf/nginx.master.conf -g daemon off;
nobody     140  0.0  0.0 152080  7460 ?        S    18:33   0:00      \_ nginx: worker process
nobody     141  0.0  0.0 152016  6212 ?        S    18:33   0:00      \_ nginx: cache manager process
nobody     142  0.0  0.0 152016  6212 ?        S    18:33   0:00      \_ nginx: cache loader process
```

We need to come up with some fixture that you can use here and also try to debug why these processes did not fail (pidfile should have made one of them to bork). This is strictly test harness related so I can take core of it for you or if you want a deeper dive into pytest - feel free to tacke it :) 

Just let me know what suits you best.